### PR TITLE
[BUG] Add a with_execution/planning_config context manager and fix tests for splitting of parquet

### PR DESCRIPTION
**Pull request by jaychia**
1. Adds a context manager to perform overrides of the configs that will reset themselves afterwards
2. Adds some fixes for splitting of Parquet files

### CODE REVIEW

**Comment by colin-ho** [comment]
added the fix here, also had to modify some of the sql tests because the scan task merging affects it

File: daft/runners/ray_runner.py [hunk 1]
```diff
@@ -350,8 +350,10 @@ def single_partition_pipeline(
     partial_metadatas: list[PartitionMetadata],
     *inputs: MicroPartition,
 ) -> list[list[PartitionMetadata] | MicroPartition]:
-    set_execution_config(daft_execution_config)
-    return build_partitions(instruction_stack, partial_metadatas, *inputs)
+    with with_execution_config(
```
Comment (colin-ho) [inline daft/runners/ray_runner.py:353]: Nit: could we change `with_execution_config` to something like `daft_execution_context` . currently the `with with_execution_config` has a double `with with` that looks a little weird on first glance

**Review by colin-ho — APPROVED** [review]
Nice catches!

File: daft/runners/ray_runner.py (hunk 1 above)
Comment (jaychia) [reply daft/runners/ray_runner.py]: Yeah I agree, changed to `*_config_ctx`

**Pull request merged**